src/app.js

```diff
@@ -42,39 +42,34 @@ app.get('/help', (req, res) => {
     })
 })
 
-app.get('/weather', (req, res) => {
-    if (!req.query.address) {
-        return res.send({
-            error: 'No address provided!'
-        })
-    }
-    geocode.geocode(req.query.address, (error, { latitude, longitude, location } = {}) => {
-        if (error) {
-            return res.send({ error })
-        }
-        forecast.forecast(latitude, longitude, (error, { description, temperature, feelslike, icon }) => {
-            if (error) {
-                return res.send({ error })
-            }
-            res.send({
-                description,
-                temperature,
-                feelslike,
-                icon,
-                location,
-                address: req.query.address
+app.get('/weather', async (req, res) => {
+    try {
+        const { address, lat, lon } = req.query
+        if (!address && !lat && !lon) {
```

Comment:
we can add better body validation, let's discuss that on next call. Example will be prepared